### DISCUSSION TITLE
PowerShell Nerd Fonts Installer

### DISCUSSION THREAD

**Discussion by jpawlowski** [ryanoasis/nerd-fonts#1697] [Show and tell]
Hello everyone,

I recently had the chance to work on a [PowerShell installer for Nerd Fonts](https://github.com/jpawlowski/nerd-fonts-installer-PS), which includes the new Cascadia Code/Mono from the Microsoft repository. 

Originally developed for a Dev Container Feature project, I believe it's worth sharing here since it has broader applications as well.

What distinguishes it is that although the script can be installed locally from the [PowerShell Gallery](https://www.powershellgallery.com/packages/Invoke-NerdFontInstaller), it is not mandatory; it can also be executed directly from the web for an immediate installation of fonts:

```powershell
& ([scriptblock]::Create((iwr 'https://to.loredo.me/Install-NerdFont.ps1')))
```

Or alternatively, without the shortened URL:

```powershell
& ([scriptblock]::Create((iwr 'https://raw.githubusercontent.com/jpawlowski/nerd-fonts-installer-PS/main/Invoke-NerdFontInstaller.ps1')))
```

If no parameters are specified (which is still possible after the above code block), a simple selection menu will be displayed:

![PowerShell Nerd Fonts Installer](https://raw.githubusercontent.com/jpawlowski/nerd-fonts-installer-PS/main/images/nerd-fonts-installer.gif)

It's also worth noting that the script utilizes the [font library file from the Nerd Font repo](https://github.com/ryanoasis/nerd-fonts/blob/master/bin/scripts/lib/fonts.json) to display the selection and support auto-completion when executed locally.

To obtain a list of all fonts and their names, you can either list them comprehensively or search for specific parts of the name:

```powershell
& ([scriptblock]::Create((iwr 'https://to.loredo.me/Install-NerdFont.ps1'))) -List All
& ([scriptblock]::Create((iwr 'https://to.loredo.me/Install-NerdFont.ps1'))) -List cas
```

Regarding security, I am aware that executing a script directly from the web can be risky. However, this practice is common in various contexts, and I believe that if the source is trustworthy and employs transport layer encryption, one might consider the convenience to outweigh the risk.

Nevertheless, please use it at your own risk; it is offered "as is." Naturally, you are welcome to review the source code and contribute pull requests on GitHub.

Best regards,
Julian

**Comment by Finii** [comment]
This looks excellent :+1:

You use the `zip` files, because Windows has no means to extract `xz` I assume? The same problem is with Homebrew, most others use the `xz`s I guess, because they are so much smaller.

Maybe it should be featured on Nerd Fonts start page?

![image](https://github.com/user-attachments/assets/e72a5124-bb7a-4c86-a1f6-92c80668c8b3)

There is already some PS script under point 6, but I guess that is rather basic and does not work with the new fonts that are not released in-repo.

Super-small issue: The script does not look at the `XDG_DATA_HOME` variable. Which I believe only 100 persons have set. But that is a Linux thing. Hmm, PS seems to run also on Mac and Linux? :-) Whatever. Linux Nerds will point out not following [XDG specs.](https://specifications.freedesktop.org/basedir-spec/latest/)

_Edit: Typo_

**Reply by jpawlowski** [reply]
> You use the `zip` files, because Windows has no means to extract `xz` I assume? The same problem is with Homebrew, most others use the `xz`s I guess, because they are so much smaller.

Indeed, it's a brilliant notion to consider alternative archives when feasible.

The XDG specs did cross my mind once, but they slipped away in the hustle of it all. Good thing there's always space for a new point version—gotta love those updates! 😉

**Reply by jpawlowski** [reply]
@Finii, your suggestions have been implemented in version `1.1.0`, which I have just published.

Newer Windows versions also come with `tar` so they will also benefit using the `xz` files.

**Reply by jpawlowski** [reply]
> Maybe it should be featured on Nerd Fonts start page?

@Finii, I have submitted a proposal through #1701.

EDIT:
~~You may have observed that I'm utilizing the bit.ly URL shortener. We could switch to a subdomain or URL path of `nerdfonts.com` to enhance the URL's trustworthiness and our/your control over it, such as `https://nerdfonts.com/install-ps` or `https://ps-install.nerdfonts.com`.~~

~~Additionally, I am willing to consider transferring the script to the NF repository.~~
